src/ttsim/interface_dag_elements/fail_if.py

```diff
@@ -276,7 +268,38 @@ def input_data_tree_is_invalid(input_data__tree: NestedData, xnp: ModuleType) ->
         ),
         tree_name="input_data__tree",
     )
-    p_id = input_data__tree.get("p_id", None)
+
+
+@fail_or_warn_function(include_if_any_element_present=["input_data__flat"])
+def input_arrays_have_different_lengths(
+    input_data__flat: FlatData,
+) -> None:
+    """Fail if the input arrays have different lengths."""
+    len_p_id_array = len(input_data__flat[("p_id",)])
+    faulty_arrays: list[str] = []
+    for key, arr in input_data__flat.items():
+        if len(arr) != len_p_id_array:
+            faulty_arrays.append(key)
+    if faulty_arrays:
+        formatted_faulty_paths = "\n".join(f"    - {p}" for p in faulty_arrays)
+        msg = format_errors_and_warnings(
+            "The lengths of the following arrays do not match the length of the `p_id` "
```

Comment:
```suggestion
            "The lengths of the following columns do not match the length of the `p_id` "
```
Let's stick to that terminology.